File: #100daysofcodewithGFG/Count Number of SubTrees having given Sum.py
def countSubtreesWithSumX(root, x):
    
    # code here
    def func(root,x,count):
        if root == None:
            return 0
        left = func(root.left,x,count)
        right = func(root.right,x,count)
        if (left+right+root.data) == x:
            count[0]+=1
        return left+right+root.data
            
    count = [0]
    ans = func(root,x,count)
    return count[0]

File: #100daysofcodewithGFG/Count Number of SubTrees having given Sum.cpp
int cnt;
int func(Node* root,int X){
    if (root == NULL){
        return 0;
    }
    int left = func(root->left,X);
    int right = func(root->right,X);
    
    if(left+right+root->data == X){
        cnt++;
    }
    return left+right+root->data;
}
int countSubtreesWithSumX(Node* root, int X)
{
	// Code here
    cnt = 0;
	int ans = func(root,X);
	return cnt;
}
